src/Service/SessionCorrelationIdService.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * Copyright 2019 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types = 1);
+
+namespace Surfnet\Tiqr\Service;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+
+final readonly class SessionCorrelationIdService
+{
+    private const SALT = 'Mr6LpJYtuWRDdVR2_7VgTChFhzQ';
+
+    public function __construct(
+        private RequestStack $requestStack
+    ) {
+    }
+
+    public function generateCorrelationId(): ?string
```

Comment:
Consider making this method more pure. In short, do not allow it to return string|null. You could add a guard to test if generating a correlation id is possible. If it is not, throw an exception. Which you can catch. Or  you could add a public precondition test method which you can call before calling the generate method.

Makes calling the method slightly more of a hassle, but makes the code cleaner in the end imho.

Reply:
It's what I had first, but indeed it makes the code that calls the function more complex. It means that constructing the logger will need a try/catch or a local variable that holds the correlation id before instantiating the logger. Because of this a null coalescing operator seemed to me the cleanest solution.

But if you think it's more important to make the method pure, I can change it.

Reply:
As discussed in real life, using both is fine with me. It depends on the design philosophy which one you pick. And in this case having the null guard in the assignment is OK with me. Otherwise you'd have to add an additional method. Adding complexity in the other place.